src/eth/TransactionManager.js

```diff
@@ -149,12 +154,47 @@ export default class TransactionManager extends PublicService {
     return minePromise;
   }
 
-  async _buildTransactionOptions(data) {
+  async _buildTransactionOptions(options, contract, method, args) {
+    const nonce = await this.get('nonce').getNonce();
+
+    if (contract && !options.gasLimit) {
+      await this._getGasLimit(options, contract, method, args, nonce);
```

Comment:
likewise here, if instead of changing things via hidden side-effects, it were just `options.gasLimit = await this._getGasLimit(...)` it would be easier to understand what's happening

Reply:
also i don't think you need to pass `nonce` because it's never read? nonce-handling could probably stay as it was

Reply:
The nonce actually needs to be passed into `estimateGas`, but if I get the nonce in `_getGasLimit` there would be some scenarios where it's left out from the final options